### ISSUE TITLE
Do not run upstream tests when only TTFs are processed

### ISSUE THREAD

**Issue by davelab6** [googlefonts/fontbakery#697]
Given the following `bakery.yaml`, where the `process_files` are only TTFs, then there should not be any source checks run, yet `upstream.yaml` exists and is as follows. 

This probably means that the web UI needs to be resilient for such repos without sources.

```
git clone https://github.com/NDSICOVER/Arima-Font.git;
cd Arima-Font/ttf/ArimaKoshi/;
cat bakery.yaml;

commit: HEAD
compiler: fontforge
license: OFL.txt
process_files: [ArimaKoshi-Black.ttf, ArimaKoshi-Bold.ttf, ArimaKoshi-ExtraBold.ttf,
  ArimaKoshi-ExtraLight.ttf, ArimaKoshi-Light.ttf, ArimaKoshi-Medium.ttf, ArimaKoshi-Regular.ttf,
  ArimaKoshi-Thin.ttf]
pyftsubset: --notdef-outline --name-IDs=* --hinting
```
...

```
Project:
  error:
  - autofix: false
    err_msg: ''
    methodDoc: Check that diacritic glyph are made completely with composite
    methodName: test_diacritic_made_as_component
    name: bakery_lint.fonttests.test_ufo
    tags: [note]
    targets: &id001 [upstream-repo]
    tool: lint
  - autofix: false
    err_msg: ''
    methodDoc: Check that diacritic glyph are made completely with mark method
    methodName: test_diacritic_made_as_mark_to_mark
    name: bakery_lint.fonttests.test_ufo
    tags: [note]
    targets: *id001
    tool: lint
  - autofix: false
    err_msg: ''
    methodDoc: Check that diacritic glyph are made completely with flat method
    methodName: test_diacritic_made_as_own_glyphs
    name: bakery_lint.fonttests.test_ufo
    tags: [note]
    targets: *id001
    tool: lint
  failure:
  - autofix: false
    err_msg: '[''COPYRIGHT.txt''] does not exist in project'
    methodDoc: Font folder should contain COPYRIGHT.txt
    methodName: test_copyrighttxt_exists
    name: bakery_lint.fonttests.test_upstream
    tags: [note]
    targets: &id002 [upstream-repo]
    tool: lint
  - autofix: false
    err_msg: '[''DESCRIPTION.en_us.html''] does not exist in project'
    methodDoc: Font folder should contain DESCRIPTION.en_us.html
    methodName: test_description_exists
    name: bakery_lint.fonttests.test_upstream
    tags: [note]
    targets: *id002
    tool: lint
  - autofix: false
    err_msg: '[''FONTLOG.txt''] does not exist in project'
    methodDoc: Font folder should contain FONTLOG.txt
    methodName: test_fontlogtxt_exists
    name: bakery_lint.fonttests.test_upstream
    tags: [note]
    targets: *id002
    tool: lint
  - autofix: false
    err_msg: '[''LICENSE.txt'', ''OFL.txt''] does not exist in project'
    methodDoc: Font folder should contain LICENSE.txt
    methodName: test_licensetxt_exists
    name: bakery_lint.fonttests.test_upstream
    tags: [required]
    targets: *id002
    tool: lint
  - autofix: false
    err_msg: File `METADATA.pb` does not exist in root of upstream repository
    methodDoc: Repository contains METADATA.pb file?
    methodName: test_metadata_pb_exists
    name: bakery_lint.fonttests.test_upstream
    tags: [note]
    targets: &id004 [upstream-repo]
    tool: lint
  fixed: []
  passed: false
  success:
  - autofix: false
    err_msg: ''
    methodDoc: Each font in family has matching glyph names?
    methodName: test_family_glyph_names_match
    name: bakery_lint.fonttests.test_upstream
    tags: [note]
    targets: &id003 [upstream-repo]
    tool: lint
  - autofix: false
    err_msg: ''
    methodDoc: Check that glyphs has same number of contours across family
    methodName: test_font_prepolation_glyph_contours
    name: bakery_lint.fonttests.test_upstream
    tags: [note]
    targets: *id003
    tool: lint
  - autofix: false
    err_msg: ''
    methodDoc: Check that glyphs has same number of points across family
    methodName: test_font_prepolation_glyph_points
    name: bakery_lint.fonttests.test_upstream
    tags: [note]
    targets: *id003
    tool: lint
  - autofix: false
    err_msg: ''
    methodDoc: Repository contains bakery.yaml configuration file?
    methodName: test_bakery_yaml_exists
    name: bakery_lint.fonttests.test_upstream
    tags: [note]
    targets: *id004
    tool: lint
  - autofix: false
    err_msg: ''
    methodDoc: Each font copyright notice matches?
    methodName: test_copyright_notices_same_across_family
    name: bakery_lint.fonttests.test_upstream
    tags: [note]
    targets: *id004
    tool: lint
```

**Comment by davelab6** [comment]
Note also that the first 3 lines of the 'small things' issue are also coming from `test_ufo.py` when they should not be:

https://github.com/googlefonts/fontbakery/issues/693

**Comment by felipesanches** [comment]
Those UFO tests that were running when not needed were addressed in https://github.com/googlefonts/fontbakery/commit/004905d4a89b888da192bd0749d3ad49b4746093

**Comment by felipesanches** [comment]
Maybe it is too premature to close this issue. There are still upstream tests being executed, but they all pass (when running fontbakery-build.py with CoinyTests where processed_files lists a single sfd file).

Maybe your use-case is different. Do you mean that things like checking for availability of a LICENSE.txt file are not needed when processed_files only has got TTF filenames declared? Would an explicit folder name in processed_files be required for running upstream-repo tests?

**Comment by davelab6** [comment]
I think 004905d is not the right fix for this, then. It is fixing the symptom, not the cause. Somehow in the pipeline process, the fontbakery-check-ttf.py file calls methods that call methods in the UFO class, and that should obviously not happen. 

I think this is a problem with the structure/design of the program, because it is too complex/sophisticated; it would be better if fontbakery-check-ttf.py was a single 'self contained' script that had everything it needs inside it. 

Similarly instead of a vertical metrics fixer that works as a tool but not as part of the build - https://github.com/googlefonts/fontbakery/issues/760#issuecomment-196521281- there should just be a fontbakery-check-ttf-family.py 'self contained' file that checks and fixes anything that requires pan-family information (like v metrics.)

**Comment by davelab6** [comment]
Closing, this will be fixed as a consequence of the 0.1.0 code design
